conans/test/functional/printer_test.py

```diff
@@ -0,0 +1,161 @@
+import unittest
+import os
+from conans.test.utils.tools import TestClient, TestServer
+
+conanfile_libA = """
+from conans import ConanFile
+
+class AConan(ConanFile):
+    name = "aaa"
+    version = "0.1"
+"""
+
+conanfile_libB = """
+from conans import ConanFile
+
+class BConan(ConanFile):
+    name = "bbb"
+    version = "0.1"
+    requires = "aaa/0.1@lasote/channel"
+"""
+
+conanfile_libC = """
+from conans import ConanFile
+
+class CConan(ConanFile):
+    name = "ccc"
+    version = "0.1"
+    requires = "bbb/0.1@lasote/channel"
+    exports = "txt_file.txt"
+
+"""
+
+txt_file = """
+    example file
+    """
+
+test_profile = """
+[settings]
+    os=Windows
+    os_build=Windows
+    arch=x86_64
+    arch_build=x86_64
+    compiler=Visual Studio
+    compiler.version=15
+    build_type=Release
+[options]
+[build_requires]
+[env]
+"""
+
+class PrinterTest(unittest.TestCase):
+
+    def setUp(self):
+        test_server = TestServer([], users={"lasote": "mypass"})
+        servers = {"default": test_server}
+
+        self.client = TestClient(servers=servers, users={"default": [("lasote", "mypass")]})
+        files = {"conanfile_libA.py": conanfile_libA,
+                 "conanfile_libB.py": conanfile_libB,
+                 "conanfile_libC.py": conanfile_libC,
+                 "txt_file.txt": txt_file,
+                 "test_profile": test_profile
+                }
+        self.client.save(files)
+        self.client.run("export conanfile_libA.py lasote/channel")
+        self.client.run("upload aaa/0.1@lasote/channel -r default")
+        self.client.run("remove -f aaa/0.1@lasote/channel")
+        self.client.run("export conanfile_libB.py lasote/channel")
+        self.client.run("upload bbb/0.1@lasote/channel -r default")
+        self.client.run("remove -f bbb/0.1@lasote/channel")
+        self.client.run("export conanfile_libC.py lasote/channel")
+
+    def print_graph_test(self):
+        self.client.run("install ccc/0.1@lasote/channel",
+                        ignore_error=True) #Ignore ERROR: Missing prebuilt package
+        self.assertIn("aaa/0.1@lasote/channel from 'default'", self.client.out)
+        self.assertIn("bbb/0.1@lasote/channel from 'default'", self.client.out)
+        self.assertIn("ccc/0.1@lasote/channel from local cache", self.client.out)
+
+    def print_info_test(self):
```

Comment:
What are you testing here? Not clear, I'd say this test is not necessary.

Reply:
Testing alll `Printer.print_info()` output. Specially for "Requires:" and "Required by:" fields depending on the reference used